lib/api/model/widget.dart

```diff
@@ -0,0 +1,131 @@
+import '../../log.dart';
+import '../../model/store.dart';
+import 'submessage.dart';
+
+/// States of a poll Zulip widget.
+///
+/// See also:
+/// - https://zulip.com/help/create-a-poll
+/// - https://github.com/zulip/zulip/blob/304d948416465c1a085122af5d752f03d6797003/web/shared/src/poll_data.ts
+class Poll {
+  Poll({
+    required this.submessages,
+    required this.pollSenderId,
+    required this.question,
+    required final List<String> options,
+  }) {
+    for (int index = 0; index < options.length; index += 1) {
+      // [null] is necessary for creating initial poll options.
+      _addOption(null, options[index], optionIndex: index);
```

Comment:
I didn't understand what this comment was trying to say until a second or third reading. Now that I do understand it: it seems like it's not conveying any information beyond "yes, saying `null` here is correct". That's not a very helpful type of comment — given that the code says `null`, the reader pretty much already knows that the code's author thought saying `null` was correct.

A question that'd be more interesting to answer is *why* the code's author (you) thought that. After seeing more of the code, I think the answer is basically that [PollEvent.optionKey] says so. So something that'd be useful to say in a comment here is to point to that.